### PR TITLE
Leanna/Shelan's Adagram

### PR DESCRIPTION
**Pull request by sheland**
# Adagrams
Congratulations! You're submitting your assignment.

## Comprehension Questions

Feature | Feedback
--- | ---
What are the components that make up a method? |
What are the advantages of using git when collaboratively working on one code base? |
What kind of relationship did you and your pair have with the unit tests? |
Does your code use any methods from the `Enumerable` mixin? If so, where and why was it helpful? |
What was one method you and your pair used to debug code? |
What are two discussion points that you and your pair discussed when giving/receiving feedback from each other that you would be willing to share? |

### CODE REVIEW

**Comment by sheland** [comment]
Our answers are in the github folder inside adagrams.rb

File: lib/adagrams.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
+require 'pry'
+require 'awesome_print'
+
+# #-------------WAVE-1--PASS-----------------
+def draw_letters
+  #string of letters of the correct quantity
+  letters = ("A" * 9) + ("B" * 2) + ("C" * 2) + ("D" * 4) + ("E" * 12) + ("F" * 2) + ("G" * 3) + ("H" * 2) + ("D" * 4) + ("E" * 12) + ("F" * 2) + ("G" * 3) + ("H" * 2) + ("I" * 9) + ("J" * 1) + ("K" * 1) + ("L" * 4) + ("M" * 2) + ("N" * 6) + ("O" * 8) + ("P" * 2) + ("Q" * 1) + ("R" * 6) + ("S" * 4) + ("T" * 6) + ("U" * 4) + ("V" * 2) + ("W" * 2) + ("Y" * 2) + ("Z" * 1)
```
Comment (CheezItMan) [inline lib/adagrams.rb:7]: Very good idea

File: lib/adagrams.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+#do not mess with this array (letters_in_hand)!!!!
+def uses_available_letters?(input, letters_in_hand)
+
+  input = input.split("") #splits the string into array
+  tru_fals = [] #array to determine validity of input later
+  input.each do |let|
+      if letters_in_hand.include?(let) #if the input letter matches a letter in hand
+        tru_fals << letters_in_hand.delete_at(letters_in_hand.index(let)) #delete letter from the one in hand & push to tru_fals
+      else
+        tru_fals << false #if no match, pushes false
```
Comment (CheezItMan) [inline lib/adagrams.rb:25]: why not just `return false` at this point?

File: lib/adagrams.rb [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+  words.each do |word|
+    word.split("")
+    word = word.to_s #turns the new array into a string
+    score = score_word(word)
+
+    if score == winner[:score] #only exceptions are ties
+
+      if winner[:word].length == 10
+        winner[:word] = winner[:word]
```
Comment (CheezItMan) [inline lib/adagrams.rb:79]: Here you're swapping the word with itself.  Seems unnecessary

File: lib/adagrams.rb [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        winner[:score] = winner[:score]
+
+      elsif word.length == 10
+        winner[:word] = word
+        winner[:score] = score
+
+      elsif word.length == 10 && winner[:word].length == 10
+        winner[:word] = winner[:word]
```
Comment (CheezItMan) [inline lib/adagrams.rb:87]: Again repetition, this `elsif` is redundant.

File: lib/adagrams.rb [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        winner[:score] = winner[:score]
+
+      elsif word.length < winner[:word].length && word.length != 10
```
Comment (CheezItMan) [inline lib/adagrams.rb:90]: I think this is redundant.

**Comment by CheezItMan** [comment]
# Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | It looks like you put them in your repo's `.PULL_REQUEST_TEMPLATE`
Both teammates contributed to the codebase | I assume so for this project
Small commits with meaningful commit messages | for the most part, however for your commit messages, use terms that describe the functionality added instead of waves completed.
**Code Requirements** |
`draw_letters` method |  Check
Uses appropriate data structure to store the letter distribution |  Very nice work here
All tests for `draw_letters` pass |  Well done
`uses_available_letters?` method | Check
All tests for `uses_available_letters?` pass |Well done
`score_word` method | Check
Uses appropriate data structure to store the letter scores |  Check
All tests for `score_word` pass |Well done
`highest_score_from` method |  Check, but see my notes, you have a bunch of redundant code here.
Appropriately handles edge cases for tie-breaking logic |
All tests for `highest_score_from` pass |Well done
**Overall** |Overall nicely done.  You have a bunch of redundant code in `highest_score_from`, but overall not bad.  See my inline notes and ask me any questions you have.  Nice work.
